src/main/java/org/gridsuite/modification/server/modifications/VscModification.java

```diff
@@ -209,46 +240,59 @@ private void hvdcAngleDroopActivePowerControlAdder(HvdcLine hvdcLine, ReportNode
-            ModificationUtils.getInstance().applyElementaryModifications(converterStation::setLossFactor, converterStation::getLossFactor, converterStationModificationInfos.getLossFactor(), converterStationReportNode, "LossFactor");
+            characteristicReports.add(ModificationUtils.getInstance().applyAndBuildModificationReport(converterStation::setLossFactor,
+                converterStation::getLossFactor, converterStationModificationInfos.getLossFactor(), "LossFactor"));
+        }
+
+        if (!characteristicReports.isEmpty()) {
+            ModificationUtils.getInstance().reportModifications(converterStationReportNode,
+                characteristicReports, "Characteristics", "Characteristics", Map.of());
         }
 
+        // set points
+        List<ReportNode> setPointsReports = new ArrayList<>();
         if (converterStationModificationInfos.getReactivePowerSetpoint() != null) {
-            ModificationUtils.getInstance().applyElementaryModifications(converterStation::setReactivePowerSetpoint, converterStation::getReactivePowerSetpoint, converterStationModificationInfos.getReactivePowerSetpoint(), converterStationReportNode, "ReactivePower");
+            setPointsReports.add(ModificationUtils.getInstance().applyAndBuildModificationReport(converterStation::setReactivePowerSetpoint,
+                converterStation::getReactivePowerSetpoint, converterStationModificationInfos.getReactivePowerSetpoint(), "Reactive Power"));
         }
 
         if (converterStationModificationInfos.getVoltageRegulationOn() != null) {
-            ModificationUtils.getInstance().applyElementaryModifications(converterStation::setVoltageRegulatorOn, converterStation::isVoltageRegulatorOn, converterStationModificationInfos.getVoltageRegulationOn(), converterStationReportNode, "VoltageRegulationOn");
+            setPointsReports.add(ModificationUtils.getInstance().applyAndBuildModificationReport(converterStation::setVoltageRegulatorOn,
+                converterStation::isVoltageRegulatorOn, converterStationModificationInfos.getVoltageRegulationOn(), "VoltageRegulationOn"));
         }
 
         if (converterStationModificationInfos.getVoltageSetpoint() != null) {
-            ModificationUtils.getInstance().applyElementaryModifications(converterStation::setVoltageSetpoint, converterStation::getVoltageSetpoint, converterStationModificationInfos.getVoltageSetpoint(), converterStationReportNode, "Voltage");
+            setPointsReports.add(ModificationUtils.getInstance().applyAndBuildModificationReport(converterStation::setVoltageSetpoint,
+                converterStation::getVoltageSetpoint, converterStationModificationInfos.getVoltageSetpoint(), "Voltage"));
+        }
+        if (!setPointsReports.isEmpty()) {
+            ModificationUtils.getInstance().reportModifications(converterStationReportNode,
+                setPointsReports, "SetPoints", "SetPoints", Map.of());
```

Comment:
```suggestion
                setPointsReports, SETPOINTS, SETPOINTS, Map.of());
```
One less upper case 'P' in `SETPOINTS` though. But I guess it is not important ?